src/Admin/Services/AccessControlService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Bit.Core.Settings;
+
+public class AccessControlService : IAccessControlService
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IConfiguration _configuration;
+    private readonly IGlobalSettings _globalSettings;
+
+    public AccessControlService(
+        IHttpContextAccessor httpContextAccessor,
+        IConfiguration configuration,
+        IGlobalSettings globalSettings)
+    {
+        _httpContextAccessor = httpContextAccessor;
+        _configuration = configuration;
+        _globalSettings = globalSettings;
+    }
+
+    public string GetUserRole(string userEmail)
+    {
+        var settings = _configuration.GetSection("adminSettings").GetChildren();
+
+        if (settings == null || !settings.Any())
+            return null;
+
+        var rolePrefix = "role";
+
+        foreach (var setting in settings)
```

Comment:
Filtering the settings to only ones that have the role prefix _could_ improve performance if most of the settings don't have the rolePrefix. However, if most of the settings have the rolePrefix, then the overhead of the filtering may outweigh the potential performance improvement. 

ex: `var filteredSettings = settings.Where(setting => setting.Key.Contains(rolePrefix) && setting.Value != null);`  // this allows for those checks to be removed from the loop.

Reply:
Expanding on your example, we can replace the bulk of the logic in that function with a linq query. Thanks!